domains/primitives/messenger/src/endpoint.rs

```diff
@@ -33,6 +42,28 @@ pub struct EndpointRequest {
     pub payload: EndpointPayload,
 }
 
+/// Request sent by src_endpoint to dst_endpoint with collected Fee
+#[derive(Debug, Encode, Decode, Clone, Eq, PartialEq, TypeInfo)]
+pub struct EndpointRequestWithCollectedFee<Balance> {
+    pub req: EndpointRequest,
+    pub collected_fee: CollectedFee<Balance>,
```

Comment:
Since only `dst_chain_fee` is needed by the dst chain we don't need to send the whole `CollectedFee`